packages/structures/src/channels/Channel.ts

```diff
@@ -0,0 +1,201 @@
+import { DiscordSnowflake } from '@sapphire/snowflake';
+import type {
+	TextChannelType,
+	ThreadChannelType,
+	APIChannel,
+	APIPartialChannel,
+	ChannelType,
+	GuildChannelType,
+	GuildTextChannelType,
+	ChannelFlags,
+} from 'discord-api-types/v10';
+import { Structure } from '../Structure.js';
+import { ChannelFlagsBitField } from '../bitfields/ChannelFlagsBitField.js';
+import { kData } from '../utils/symbols.js';
+import type { ChannelPermissionMixin } from './mixins/ChannelPermissionMixin.js';
+import type { ChannelWebhookMixin } from './mixins/ChannelWebhookMixin.js';
+import type { DMChannelMixin } from './mixins/DMChannelMixin.js';
+import type { GuildChannelMixin } from './mixins/GuildChannelMixin.js';
+import type { TextChannelMixin } from './mixins/TextChannelMixin.js';
+import type { ThreadChannelMixin } from './mixins/ThreadChannelMixin.js';
+import type { ThreadOnlyChannelMixin } from './mixins/ThreadOnlyChannelMixin.js';
+import type { VoiceChannelMixin } from './mixins/VoiceChannelMixin.js';
+
+export type PartialChannel = Channel<ChannelType, Exclude<keyof APIChannel, keyof APIPartialChannel>>;
+
+/**
+ * @internal
+ */
+export type ChannelDataType<Type extends ChannelType> = Type extends ChannelType
+	? Extract<APIChannel, { type: Type }>
+	: never; // TODO: this breaks without a conditional, find out why
+
+/**
+ * Represents any channel on Discord.
+ *
+ * @typeParam Type - Specify the type of the channel being constructed for more accurate data types
+ * @typeParam Omitted - Specify the properties that will not be stored in the raw data field as a union, implement via `DataTemplate`
+ * @remarks Although this class _can_ be instantiated directly for any channel type,
+ * it's intended to be subclassed with the appropriate mixins for each channel type.
+ */
+export class Channel<
+	Type extends ChannelType = ChannelType,
```

Comment:
This needs the `| 'unknown'` to allow future channel types to not break and be useable (which makes the TODO comment above irrelevant cause you'll need the conditional anyways)

I think it should be the default but there's probably some good reasons for it to not be

Reply:
APIPartialChannel I guess, the only thing different is the type guarantee and flags....and given that you have to skirt type safety on this anyways it's probably fine?